tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -53,6 +53,83 @@ static void check_out_type_matches(Tensor result,
   }
 }
 
+inline Tensor & dispatch_arange(Scalar end, Tensor result) {
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(result);
+  return at::arange_out(result, end);
+}
+
+inline Tensor dispatch_arange(Scalar end, const Type & dtype, int64_t device) {
+  maybe_initialize_cuda(dtype);
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(device);
+  return at::arange(dtype, end);
+}
+
+inline Tensor & dispatch_arange(Scalar start, Scalar end, Scalar step, Tensor result) {
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(result);
+  return at::arange_out(result, start, end, step);
+}
+
+inline Tensor dispatch_arange(Scalar start, Scalar end, Scalar step, const Type & dtype, int64_t device) {
+  maybe_initialize_cuda(dtype);
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(device);
+  return at::arange(dtype, start, end, step);
+}
+
+static inline bool allIntegral(std::initializer_list<Scalar> l) {
+  for (auto s : l) {
```

Comment:
nit: use a reference? scalars can be backed by tensors after all

Reply:
Looks like I have to use `std::initializer_list<std::reference_wrapper<Scalar>>`